salt/cloud/clouds/aliyun.py

```diff
@@ -770,7 +770,7 @@ def percent_encode(line):
     # All aliyun API only support GET method
     stringToSign = 'GET&%2F&' + percent_encode(canonicalizedQueryString[1:])
 
-    h = hmac.new(access_key_secret + "&", stringToSign, sha1)
+    h = hmac.new(bytearray(access_key_secret + "&", 'utf-8'), bytearray(stringToSign, 'utf-8'), sha1)
```

Comment:
While I can't test this change, I think it would be better to use the `salt.utils.stringutils` functions instead of hard-coding `utf-8` here.

Reply:
Maybe i shuold move this change to develop branch ? (This is my first pr)
i will use `salt.utils.stringutils` functions instead of `utf8`

Thx:blush:

Reply:
Sure, that would be fine. Thank you so much for contributing. We are so happy to have you. :)